cppsdk/include/playfab/PlayFabEventRouter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#ifndef DISABLE_PLAYFABENTITY_API
+
+#include <playfab/PlayFabEvent.h>
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Interface for any event router
+    /// </summary>
+    class IPlayFabEventRouter
+    {
+    public:
+        virtual ~IPlayFabEventRouter() {}
+        virtual void RouteEvent(std::shared_ptr<const IPlayFabEmitEventRequest> request) const = 0; // Route an event to pipelines. This method must be thread-safe.
+
+    private:
+        //std::unordered_map<size_t, std::shared_ptr<IPlayFabPipeline>> pipelines;
```

Comment:
>//std::unordered_map<size_t, std::shared_ptr<IPlayFabPipeline>> pipelines; [](start = 8, length = 74)

not needed ?

Reply:
It will be a part of the next chunk of work, but i had to comment it out for two reasons:
- to show the intent of this class
- to make it actually compilable while the other components of story are not implemented yet

---
In reply to: [227947899](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227947899) [](ancestors = 227947899)